clients/cobol-lsp-vscode-extension/src/services/copybook/downloader/DownloadUtil.ts

```diff
@@ -72,16 +72,31 @@ export class DownloadUtil {
   public static async checkForInvalidCredProfile(
     profileName: string,
     explorerAPI: IApiRegisterClient,
+    documentUri: string,
+    copybookNames: CopybookName[],
   ): Promise<boolean> {
     if (
       ZoweExplorerDownloader.profileStore.get(profileName) === "valid-profile"
     ) {
       return false;
     }
 
+    const copybookLocation = this.areCopybookDownloadConfigurationsPresent(
+      documentUri,
+      copybookNames,
+    );
+
+    if (!copybookLocation) {
+      return true;
+    }
+
     try {
       const profile = this.loadProfile(profileName, explorerAPI);
-      await explorerAPI.getUssApi(profile).fileList("/");
+      if (copybookLocation.uss) {
+        await explorerAPI.getUssApi(profile).fileList(copybookLocation.uss);
+      } else if (copybookLocation.dsn) {
+        await explorerAPI.getMvsApi(profile).allMembers(copybookLocation.dsn);
+      }
```

Comment:
Should we check just for invalid creds error?

Reply:
All errors are handled in the check, just the invalid credentials error gets special treatment and shows error message to the user. Other errors are just logged into the telemetry service. I searched the telemetry data if there are some other unhandled errors reported, but I haven't found any.

Reply:
But we still end up reporting that the profile has invalid credential if e.g. the dataset does not exist.

Reply:
I investigated the explorer API and added checks for NotFound error (dataset/uss folder doesn't exists) and Authorization error (user doesn't have permission to access the dataset). For these situations the credentials are considered as valid and copybooks can be downloaded.